ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JClient.java

```diff
@@ -67,7 +60,12 @@ public <T> SafeFuture<Response<T>> doRequest(
         .handle(
             (response, exception) -> {
               if (exception != null) {
-                handleError(exception);
+                final boolean authError =
+                    exception instanceof ClientConnectionException
+                        && exception.getMessage() != null
+                        // Could be different authorization error codes, depends on client
+                        && exception.getMessage().contains("received: 40");
```

Comment:
This would include 404 errors which shouldn't be considered an auth error. In theory it should always be a 401 Unauthorised if authentication fails - unless I'm missing something else relevant.  I'd be tempted to just check for that explicitly given this only affects logging and then raise issues with clients returning a different code.  Possibly something that should be specified in the execution engine API spec as well.

Reply:
I saw both 401 and 403 errors with different ELs, wanted to join it, but you are definitely right about 404. I will limit it to 401 and 403